### ISSUE TITLE
support JGI and EMSL project identifiers that aren't CURIES registered with identifiers.org 

### ISSUE THREAD

**Issue by aclum** [microbiomedata/nmdc-schema#868]
We have use cases from JGI, EMSL and external projects where we need to point to data/projects/analysis where the base url is not registered as a CURIE with identifiers.org

JGI genome portal proposal link (where JGI proposal ID is 502930)
https://genome.jgi.doe.gov/portal/lookup?keyName=proposalId&keyValue=502930&groupOnly=1&app=Info, this resolves to https://genome.jgi.doe.gov/portal/BioDefcarcycling/BioDefcarcycling.info.html

This is an example of a possible valid slot for a JGI genome portal proposal, analogous to GOLD study, url.
https://github.com/microbiomedata/nmdc-schema/blob/jgi_genome_portal_id/src/data/valid/Study-genome_portal_identifiers.yaml

Processed metabolomics data in GNPS
https://gnps.ucsd.edu/ProteoSAFe/status.jsp?task=4b848c342a4f4abc871bdf8a09a60807

EMSL examples
study example https://www.emsl.pnnl.gov/project/60141 
samples and data https://search.emsl.pnnl.gov/?project_id_search=60141 

cc @mslarae13 

As a starting point we could focus how to do this for class Study. The JGI genome portal schema support is important for the upcoming bioscales publication since the paper cites some data that aren't in GOLD.

needed work:
- [x] register CURIES
- [x] Add slot for GNPS identifiers to NMDC schema
- [x] Add slot for JGI genome portal study 

A slot for EMSL study curie is pending discussions

**Comment by cmungall** [comment]
Here is the SOP

To go:

https://github.com/biopragmatics/bioregistry/issues/new/choose

Fill in the fields as best you can. Ideally rearrange the order of http params so the local ID comes last, e.g. 

https://genome.jgi.doe.gov/portal/lookup?keyName=proposalId&groupOnly=1&app=Info&keyValue=$1

I suggest the following prefixes:

* `jgi.proposal`
* `emsl.study`
* `gnps`

Here is an example of how you could fill this in for `jgi.proposal`

 * prefix `jgi.proposal`
 * regex `^\d+`
 * local id: 502930
 * URI format string: https://genome.jgi.doe.gov/portal/lookup?keyName=proposalId&groupOnly=1&app=Info&keyValue=$1

I recommend tagging the person in charge of the JGI proposal database that we are doing this, same for others

**Comment by aclum** [comment]
Feedback from Ming on prefixes

gnps.task:<uuid> # these are analyses
gnps.library:<library name> # these are collections of spectra that are reference data
gnps.libraryentry:<library accession> # these are individual MS/MS spectra that are part of libraries

**Comment by cmungall** [comment]
- https://github.com/biopragmatics/bioregistry/issues/833

**Comment by cthoyt** [comment]
Chris and I got the `gnps.task` and `emsl.project` prefixes into the Bioregistry in https://github.com/biopragmatics/bioregistry/pull/836 and https://github.com/biopragmatics/bioregistry/pull/834, respectively. These will be available momentarily (following database rebuild and re-deploy) at https://bioregistry.io/gnps.task and https://bioregistry.io/emsl.project

**Comment by cthoyt** [comment]
Anyone who is interested to add more prefixes to the Bioregistry can feel free to reach out to me directly or open an issue on the tracker like Chris mentioned

**Comment by aclum** [comment]
@mslarae13  what is your suggested prefix for the 'samples and data' search? https://search.emsl.pnnl.gov/?project_id_search=60141

**Comment by cmungall** [comment]
I think that is the same object, it's a study, so no need for a separate prefix. You could register this as a separate resolver for a study but I don't think it's necessary as the samples are linked from the study page

**Comment by aclum** [comment]
This is actively being worked on, once the curies are registered we'll need schema slots for these. I'll update my branch with valid data based on the newly defined curies.

**Comment by cmungall** [comment]
OK, we have all 3 categories - I think we can close this now?

**Comment by mslarae13** [comment]
I'd like to discuss this at the 1pm sync today (05/24)
@cmungall , @aclum

**Comment by cthoyt** [comment]
Can we try and make a slide deck that explains the workflow that happened here?

I'd like to get a conceptual overview of what happened here, both 1) for people working on LinkML projects and 2) to have a high-level use case that we can communicate for the utility of the Bioregistry. 

Specifically, I think it would be great to summarize:

1. How does a project using LinkML determine it needs a new prefix
2. How does a project request a new prefix (okay, I know how this works ;))
3. What does the project do after the new prefixes are available?

can someone work with me to make a brief slide deck or google doc describing this? I guess this would also be useful to include as a tutorial in the LinkML documentation, right?

**Comment by turbomam** [comment]
> @turbomam can you update range for gnps_identifiers to be curie instead of string?

I just changed all alternative_identifiers to have a uriorcurie range in src/schema/basic_slots.yaml. I'm rebuilding and testing now. I got an error and am invesitgating.

We may also want to apply a pattern constraint

We should check for how those propagate to sub-properties. I have seen some attributes of `id` not get propagated, but that may have been though some convoluted paths.

I feel like @sujaypatil96 proposed doing the same thing I'm doing now, but we hesitated to merge the PR. I'll try to dig that up now.
